### PR TITLE
Update schema

### PR DESCRIPTION
**Pull request by michalsidzej**

### CODE REVIEW

File: packages/backend/src/peripherals/database/types.ts [hunk 1]
```diff
@@ -10,12 +10,14 @@ declare module 'knex/types/tables' {
 
   interface PriceRow {
     coingecko_id: string
+    asset_id?: string
```
Comment (sin6pi7) [inline packages/backend/src/peripherals/database/types.ts:13]: Why is this optional?

Comment (michalsidzej) [reply packages/backend/src/peripherals/database/types.ts:13]: I just wanted to stay without TS errors everywhere

File: packages/backend/src/peripherals/database/types.ts [hunk 2]
```diff
@@ -10,12 +10,14 @@ declare module 'knex/types/tables' {
...
     price_usd: number
     unix_timestamp: string
   }
 
   interface BalanceRow {
     block_number: number
+    unix_timestamp?: string
```
[the remark this answers is not in the document]
Comment (michalsidzej) [reply packages/backend/src/peripherals/database/types.ts:20]: jw

File: packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
       return false
     }
     return true
   })
 }
 
-function getBridges(projects: ProjectInfo[]) {
+// change
+// now tokens are filtered based on project not bridge
+// todo discuss it
```
Comment (sin6pi7) [inline packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts:23]: Do we have a conclusion here?

File: packages/backend/src/core/ReportUpdater.ts [hunk 4]
```diff
@@ -58,20 +80,64 @@ export class ReportUpdater {
...
+    )
+
+    console.log('balancesPerProject', balancesPerProject)
+
     const reports: ReportRecord[] = []
-    for (const balance of balances) {
+    for (const balance of balancesPerProject) {
       const token = this.tokenByAssetId.get(balance.assetId)
       if (!token) {
+        console.log('dupa', balance.assetId, this.tokenByAssetId)
```
Comment (sin6pi7) [inline packages/backend/src/core/ReportUpdater.ts:94]: I guess these `console.log`s are not supposed to be here 😆

File: packages/backend/src/peripherals/database/migrations/013_update_balances.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export async function up(knex: Knex) {
+  await knex.schema.alterTable('asset_balances', (table) => {
+    table.bigInteger('unix_timestamp')
+  })
+  await knex.raw(`
+    UPDATE asset_balances
+    SET unix_timestamp = (
+      SELECT unix_timestamp
+      FROM block_numbers
+      WHERE asset_balances.block_number = block_numbers.block_number
```
Comment (sin6pi7) [inline packages/backend/src/peripherals/database/migrations/013_update_balances.ts:25]: Does this work as expected?

File: packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts (hunk 3 above)
Comment (michalsidzej) [reply packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts:23]: Let's raise it here: We are now filtering the reports based on projects, not bridge. In an event that our config changes in a way, that one of the bridges (b1) stops tracking a particular asset AND other bridge does track it, this filter will not discard reports from b1. I guess this is an unlikely event, so it shouldn't be a problem to keep it as it is for now.

File: packages/backend/src/peripherals/database/migrations/013_update_balances.ts (hunk 5 above)
Comment (michalsidzej) [reply packages/backend/src/peripherals/database/migrations/013_update_balances.ts:25]: Yes, updates unix_timestamp column with a on block_number equality

File: packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts (hunk 3 above)
Comment (sz-piotr) [inline packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts:23]: I think that ultimately there should be no or minimal filtering done. The filtering should occur in ReportUpdater where non-conforming records are deleted from the database

File: packages/backend/src/api/controllers/report/filter/getSufficientlySynced.ts [hunk 6]
```diff
@@ -1,16 +1,16 @@
 import { ReportRecord } from '../../../../peripherals/database/ReportRepository'
-import { getMaxAssetInBridge } from './getMaxAssetsInBridge'
+import { getNewestAssetsInProject } from './getNewestAssetsInProject'
 import { getSyncedTimestamp } from './getSyncedTimestamp'
 
 export function getSufficientlySynced(reports: ReportRecord[]) {
```
Comment (sz-piotr) [inline packages/backend/src/api/controllers/report/filter/getSufficientlySynced.ts:5]: Again with the filtering I think that in the future this logic should just be removed. As long as ReportUpdater leaves only relevant records it's fine.

File: packages/backend/src/core/ReportUpdater.ts [hunk 7]
```diff
@@ -14,37 +20,53 @@ import {
...
   ) {
     this.logger = this.logger.for(this)
     for (const token of this.tokens) {
       this.tokenByAssetId.set(token.id, token)
     }
+
+    for (const { projectId, bridges } of projects) {
+      this.projectDetailsById.set(projectId, {
+        bridges: bridges.map((b) => EthereumAddress.unsafe(b.address)),
+        assetIds: bridges.flatMap((b) => b.tokens.map((t) => t.id)),
```
Comment (sz-piotr) [inline packages/backend/src/core/ReportUpdater.ts:51]: This should probably be deduplicated

File: packages/backend/src/peripherals/database/BlockNumberRepository.ts [hunk 8]
```diff
@@ -35,6 +35,16 @@ export class BlockNumberRepository extends BaseRepository {
     return rows.map(toRecord)
   }
 
+  async findByBlockNumber(blockNumber: bigint): Promise<BlockNumberRecord> {
+    const knex = await this.knex()
+    const [row] = await knex('block_numbers').where(
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/BlockNumberRepository.ts:40]: You should use .first() and hadle undefined

**Review by sz-piotr — CHANGES_REQUESTED** [review]

**Review by sz-piotr — APPROVED** [review]

**Pull request merged**